src/Language/PureScript/TypeChecker/Subsumption.hs

```diff
@@ -112,14 +111,22 @@ subsumes' mode (TypeApp _ f1 r1) (TypeApp _ f2 r2) | eqType f1 tyRecord && eqTyp
       (for_ (firstMissingProp ts2' ts1') (throwError . errorMessage . PropertyIsMissing . rowListLabel))
     when (eqType r2' $ REmpty ())
       (for_ (firstMissingProp ts1' ts2') (throwError . errorMessage . AdditionalProperty . rowListLabel))
+    ts1'' <- traverse removePolyTypes ts1'
+    ts2'' <- traverse removePolyTypes ts2'
     -- Check subsumption for common labels
     sequence_ common
-    unifyTypes (rowFromList (ts1', r1')) (rowFromList (ts2', r2'))
+    unifyTypes (rowFromList (ts1'', r1')) (rowFromList (ts2'', r2'))
     -- Nothing was elaborated, return the default coercion
     return (defaultCoercion mode)
   where
     -- Find the first property that's in the first list (of tuples) but not in the second
     firstMissingProp t1 t2 = fst <$> uncons (minusBy' (comparing rowListLabel) t1 t2)
+    mapRowListItem :: Monad m => (Type a -> m (Type a)) -> RowListItem a -> m (RowListItem a)
+    mapRowListItem f rli = do
+      ty' <- f (rowListType rli)
+      return $ rli { rowListType = ty' }
+    removePolyTypes :: MonadState CheckState m => RowListItem SourceAnn -> m (RowListItem SourceAnn)
+    removePolyTypes = mapRowListItem instantiatePolyTypeWithUnknowns
```

Comment:
This is going to instantiate constraints and not pass in the appropriate dictionaries. I'd encourage you to write down the elaboration rules you're implementing here. I don't think there's a good way to support constraints.